pkg/interop/iterator/iterator.go

```diff
@@ -18,6 +18,15 @@ func Create(items []interface{}) Iterator {
 	return Iterator{}
 }
 
+// Concat concatenates two given iterators returning one that will range on
+// a first and then continue with b. Iterator positions are not reset for a
+// and b, so if any of them was already advanced by Next the resulting
+// Iterator will point at this new position and never go back to previous
+// key-value pairs. This function uses `Neo.Iterator.Concat` syscall.
+func Concat(a, b Iterator) Iterator {
```

Comment:
Another thing that can be important, that `Key()` call on an concatenated iterator will return "internal key" of the one of concatenated iterators, so there can be 2 items with the same key.
https://github.com/neo-project/neo/blob/master-2.x/neo.UnitTests/UT_ConcatenatedIterator.cs#L54

Reply:
Added a note.